src/archive_librarian/dbHandler.cpp

```diff
@@ -27,78 +31,151 @@
+#define UPDATE_SCHEMA(details)\
+    char* err; \
+    if (sqlite3_exec(db_, details.c_str(), nullptr, nullptr, &err) != SQLITE_OK) { \
+        printf("Error: Failed to execute schema update: %s\n", err); \
+        sqlite3_free(err); \
+        ROLLBACK_AND_RETURN(); \
+    }
+*/
+
 // -------------------------
 // DBHandler Constructor / Destructor
 // -------------------------
 
-DBHandler::DBHandler(const std::string& schemaSQL, const std::string& dbPath, size_t maxCacheSize)
-    : jobIdCache_(maxCacheSize)
-{
-    if (sqlite3_open(dbPath.c_str(), &db_) != SQLITE_OK) {
+bool DBHandler::initialize() {
+    jobIdCache_.setLimit((size_t)(config[conf::i::DBMaxJobCacheSize]));
+
+    if (sqlite3_open(config[conf::str::DBPath].c_str(), &db_) != SQLITE_OK) {
         std::string err = sqlite3_errmsg(db_);
+        dprintf(D_ERROR, "Failed to open SQLite Data Base: %s\n", err.c_str());
         sqlite3_close(db_); // technically redundant if db is already bad, but harmless
-        throw std::runtime_error("Failed to open SQLite DB: " + err); // TODO CHANGE THIS
+        return false;
     }
 
-    if (!initializeFromSchema(schemaSQL)) {
-        sqlite3_close(db_);
-        db_ = nullptr;
-        throw std::runtime_error("Failed to initialize schema");
+    char* errMsg = nullptr;
+    int rc;
+
+    // Enable foreign key constraints
+    rc = sqlite3_exec(db_, "PRAGMA foreign_keys = ON;", nullptr, nullptr, &errMsg);
+    if (rc != SQLITE_OK) {
+        dprintf(D_ERROR, "Failed to enable foreign keys: %s\n", errMsg ? errMsg : "Unknown");
+        sqlite3_free(errMsg);
+        return false;
+    }
+
+    int version = getSchemaVersion();
+    if (version == -1) {
+        dprintf(D_ERROR, "Failed to get current schema version.\n");
+        return false;
+    }
+
+    dprintf(D_ALWAYS, "Database Schema Version: %d\n", version);
+
+    // Begin transaction for atomic schema creation
+    rc = sqlite3_exec(db_, "BEGIN IMMEDIATE TRANSACTION;", nullptr, nullptr, &errMsg);
+    if (rc != SQLITE_OK) {
+        dprintf(D_ERROR, "Failed to begin transaction: %s\n", errMsg ? errMsg : "Unknown");
+        sqlite3_free(errMsg);
+        return false;
     }
 
+    // Execute the schema SQL
+    rc = sqlite3_exec(db_, SavedQueries::SCHEMA_SQL.c_str(), nullptr, nullptr, &errMsg);
+    if (rc != SQLITE_OK) {
+        dprintf(D_ERROR, "Failed to initialize DB schema: %s\n", errMsg ? errMsg : "Unknown");
+        sqlite3_free(errMsg);
+        ROLLBACK_AND_RETURN();
+    }
+
+    // Switch to run version modification statments of database
+    // Note: Use case fall throughs adding newer versions lower
+    //       Also, don't forget to change the new version in the database!!!
+    switch (version) {
+        case 0: // Initializing new database
+            [[fallthrough]];
+        default:
+            if (sqlite3_exec(db_, "PRAGMA user_version = 1;", nullptr, nullptr, &errMsg) != SQLITE_OK) {
```

Comment:
Would it make sense for the `version` value and the value to which `user_version` is set match?